### PR TITLE
contribute a repo query tool and scripts

### PR DESCRIPTION
**Pull request by devoncarew**
- contribute a repo query tool and scripts

```
Run various reports on Dart and Flutter related repositories.

Usage: report <command> [arguments]

Global options:
-h, --help    Print this usage information.

Available commands:
  branches   Show the default branch names of Dart and Flutter repos.
  labels     Report on the various labels in use by dart-lang repos.
  links      List useful GitHub query urls.
  weekly     Run a week-based report on repo status and activity.

Run "report help <command>" for more information about a command.
```

### CODE REVIEW

File: pkgs/repo_query/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: repo_query
+description: Miscellaneous issue, repo, and PR query tools.
+
+publish_to: none
+
+environment:
+  sdk: ^2.19.0
+
+dependencies:
```
Comment (kevmoo) [inline pkgs/repo_query/pubspec.yaml:9]: Dude: there's pkg:github!

Comment (devoncarew) [reply pkgs/repo_query/pubspec.yaml:9]: I've not used it before; sounds like I should take a look (esp. if it can lop off ~100 lines of code here).

Comment (kevmoo) [reply pkgs/repo_query/pubspec.yaml:9]: Yeah, like don't hand-write JSON decoding logic!

Comment (devoncarew) [reply pkgs/repo_query/pubspec.yaml:9]: updated -

**Comment by devoncarew** [comment]
@vsmenon - here's the script(s) I used to build that repo stats spreadsheet

File: pkgs/repo_query/lib/labels.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    for (var entry in results.entries) {
+      var repo = entry.key;
+      if (repo.openIssuesCount < 30) continue;
+
+      for (var label in entry.value) {
+        var labelInfo =
+            labels.putIfAbsent(label.name, () => _LabelInfo(label.name));
+
+        labelInfo.repos.add(repo.name);
+        labelInfo.weight += log(repo.openIssuesCount);
```
Comment (mosuem) [inline pkgs/repo_query/lib/labels.dart:46]: Maybe extract the two occurrences of `repo.openIssuesCount` to a local variable, to make the previous check for `openIssuesCount<30` more clear?

File: pkgs/repo_query/lib/src/common.dart [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    required this.issuesOpened,
+    required this.issuesClosed,
+    required this.commits,
+    required this.p0Count,
+    required this.p1Count,
+    required this.stargazers,
+  });
+}
+
+// This are monorepos, high-traffic repos, or otherwise noteable repos.
```
Comment (mosuem) [inline pkgs/repo_query/lib/src/common.dart:133]:
```suggestion
// These are monorepos, high-traffic repos, or otherwise noteable repos.
```

**Review by mosuem — APPROVED** [review]

File: pkgs/repo_query/lib/labels.dart (hunk 2 above)
Comment (devoncarew) [reply pkgs/repo_query/lib/labels.dart]: done! (and, added a comment about the issue count cutoff)

**Pull request merged**